executor/executor.go

```diff
@@ -0,0 +1,235 @@
+// Package executor implements a high level execution context with monitoring and
+// logging features.
+//
+// Please see Executor for more information.
+package executor
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os/exec"
+	"syscall"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+)
+
+// ExecResult is the result of a Stop() operation and contains various fields
+// related to the post-mortem state of the process such as output and exit
+// status.
+type ExecResult struct {
+	Stdout     string
+	Stderr     string
+	ExitStatus uint32
+	Runtime    time.Duration
+
+	executor *Executor
+}
+
+// Executor is the context used to execute a process. The runtime state is kept
+// here. Please see the struct fields for more information.
+//
+// New() is the appropriate way to initialize this type.
+//
+// No attempt is made to manage concurrent requests to this struct after
+// the program has started.
+type Executor struct {
+	// The command passed in through New()
+	Command *exec.Cmd
```

Comment:
I think if it is not expected to change once `New()` is called, we should not export this. Also that way we will also force the caller to get things stdout/stderr from `ExecResult` than from `e.Command`

Similar for fields below as well. I think making them exported would make behavior a bit unsafe/anomalous if they are changed after call to `Start()` or may be we should just add comments stating that these should not be changed once command is started.